consensus/testing.go

```diff
@@ -33,28 +31,73 @@ type TestPowerTableView struct {
 	minerToWorker          map[address.Address]address.Address
 }
 
+// NewTestActorState provides a queryer that responds to power table view queries wih the given parameters
+type TestActorState struct {
+	minerPower    *types.BytesAmount
+	totalPower    *types.BytesAmount
+	minerToWorker map[address.Address]address.Address
+}
+
+func NewTestActorState(minerPower, totalPower *types.BytesAmount, minerToWorker map[address.Address]address.Address) *TestActorState {
+	return &TestActorState{
+		minerPower:    minerPower,
+		totalPower:    totalPower,
+		minerToWorker: minerToWorker,
+	}
+}
+
+func (t *TestActorState) stateTreeQueryer(st state.Tree, bh *types.BlockHeight) ActorStateQueryer {
+	return &TestQuerier{
+		minerPower:    t.minerPower,
+		totalPower:    t.totalPower,
+		minerToWorker: t.minerToWorker,
+	}
+}
+
+type TestQuerier struct {
+	minerPower    *types.BytesAmount
+	totalPower    *types.BytesAmount
+	minerToWorker map[address.Address]address.Address
+}
+
+func (tq *TestQuerier) Query(ctx context.Context, optFrom, to address.Address, method string, params ...interface{}) ([][]byte, error) {
+	if method == "getTotalStorage" {
+		return [][]byte{tq.totalPower.Bytes()}, nil
+	} else if method == "getPower" {
+		// always return 1
+		return [][]byte{tq.minerPower.Bytes()}, nil
+	} else if method == "getWorker" {
+		if tq.minerToWorker != nil {
+			return [][]byte{tq.minerToWorker[to].Bytes()}, nil
+		}
+		// just return the miner address
```

Comment:
This behavior is a bit unexpected.  It would make this method more easy to predict (though a bit more burdensome on the user) if this special casing were not here.  If you keep this it would be good to call it out explicitly in a comment over the TestQuerier.

Reply:
Good call. It was inherited from the old test code, but was not actually being used. Deleted.